src/pages/game/scenes/MainGame.mjs

```diff
@@ -236,6 +241,37 @@ export default {
 	create() {
 		MainGameScene = this;
 
+		// Modal state tracking for first-time notifications
+		this._modalState = {
+			romeFound: false,
+			firstFoodArrived: false,
+			lastFarmWarningTs: 0,
+			// Spoilage aggregation state
+			_spoilBuffer: [],
+			_spoilTimer: null,
+			_lastSpoilNotifyTs: 0,
+		};
+
+		// Global first-enemy detection: trigger once when any enemy unit
+		// is revealed on the map (not just when Rome is found).
+		currentGame.events.on('hex-visible', (evt) => {
```

Comment:
As with tracking the modal states, let's move all of these event listeners to `Modal.mjs` so that all of the logic that decides when to display which modal and how is kept in one file.